src/lib/proof-system/zkprogram.ts

```diff
@@ -1266,10 +1327,22 @@ function picklesRuleFromFunction(
     if (publicInputType === Undefined || publicInputType === Void) {
       result = (await func(...finalArgs)) as any;
     } else {
-      let input = fromFieldVars(publicInputType, publicInput);
+      console.log('auxData before input', auxInputData);
+      let input = fromFieldVars(publicInputType, publicInput, auxInputData);
       result = (await func(input, ...finalArgs)) as any;
     }
 
+    console.log('result input', result);
+    if (result.publicOutput) {
```

Comment:
this should probably be `result?.publicOutput` since `result` is unfornately typed